test/integration/instance.test.js

```diff
@@ -898,6 +898,38 @@ describe(Support.getTestDialectTeaser('Instance'), function() {
       });
     });
 
+    it('handles an entry with primaryKey of zero', function() {
+      var username = 'user'
+        , newUsername = 'newUser'
+        , User2 = this.sequelize.define('User2',
+          {
+            id: {
+              type: DataTypes.INTEGER.UNSIGNED,
+              autoIncrement: false,
+              primaryKey: true
+            },
+            username: { type: DataTypes.STRING }
+          });
+
+      return User2.sync().then(function () {
+        return User2.create({id: 0, username: username}).then(function (user){
+          expect(user).to.be.ok;
+          expect(user.id).to.equal(0);
+          expect(user.username).to.equal(username);
+          return User2.find(0).then(function (user) {
```

Comment:
Yep - the 0 is ignored by find, it only works because there is only a single instance :)